### PR TITLE
feat(core/dropdown-button): add dropdown-button

### PR DESCRIPTION
**Pull request by goncalosard**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Unit tests (`yarn test`) were run locally and passed
- [ ] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [ ] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bug fix
- [x] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the new behavior?

Implemented new component dropdown-button as requested in #173 

image

A new prop in icon-button was also implemented to let the small triangle to be toggle when its used on a dropdown-button.

Build is not passing because of the problem that i'm Windows, so the file component-doc.json needs to be re-commited by someone with mac.

Visual-regression tests for this new component are created, but it is needed to check if any onther tests are failing and the new 
 snapshots need to be commited.

## Code used for testing

        <ix-dropdown-button
          label="Dropdown"
          variant="Primary"
          icon="checkboxes"
        ></ix-dropdown-button>
        <ix-dropdown-button
          label="Dropdown"
          variant="Primary"
          outline
          icon="checkboxes"
        ></ix-dropdown-button>
        <ix-dropdown-button
          label="Dropdown"
          variant="Primary"
          ghost
          icon="checkboxes"
        ></ix-dropdown-button>
        <ix-dropdown-button
          label="Dropdown"
          variant="Primary"
          disabled
          icon="checkboxes"
        ></ix-dropdown-button>
        <ix-dropdown-button
          label=""
          variant="Primary"
          icon="checkboxes"
        ></ix-dropdown-button>
        <ix-dropdown-button
          label=""
          variant="Primary"
          outline
          icon="checkboxes"
        ></ix-dropdown-button>
        <ix-dropdown-button
          label=""
          variant="Primary"
          ghost
          icon="checkboxes"
        ></ix-dropdown-button>
        <ix-dropdown-button
          label=""
          variant="Primary"
          disabled
          icon="checkboxes"
        ></ix-dropdown-button>

## Testing

Tested locally

### CODE REVIEW

**Review by danielleroux — CHANGES_REQUESTED** [review]
- documentation is missing
- angular examples are missing
- react examples are missing
- http examples are missing

File: packages/core/src/components/dropdown-button/dropdown-button.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Siemens AG
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+.hide {
```
Comment (danielleroux) [inline packages/core/src/components/dropdown-button/dropdown-button.scss:10]: add :host selector around class

File: packages/core/src/components/icon-button/icon-button.tsx [hunk 2]
```diff
@@ -74,6 +74,11 @@ export class IconButton implements Button {
    */
   @Prop() type: 'button' | 'submit' = 'button';
 
+  /**
+   * Enable dropdown button triangle
+   */
+  @Prop() dropdownTriangle = false;
+
```
Comment (danielleroux) [inline packages/core/src/components/icon-button/icon-button.tsx:81]: We should not add the `dropdown-triangle` as feature flag to icon button.

Please implement the dropdown icon in dropdown button itself.

**Review by danielleroux — CHANGES_REQUESTED** [review]

**Review by danielleroux — CHANGES_REQUESTED** [review]
- DropdownButton should also provide a possiblity to open a dropdown e.g.

```
<ix-dropdown-button>
  <ix-dropdown-item></ix-dropdown-item>
  <ix-dropdown-item></ix-dropdown-item>
  ....
</ix-dropdown-button>
```

See ix-select

**Comment by goncalosard** [comment]
@danielleroux 
Last commit fixed the requests you made(css, triangle, dropdown). Added documentation and examples. 
Triangle color is not correct, it cannot inherit, so a variable is needed to match the colors of the icons with the triangle.

**Comment by danielleroux** [comment]
@goncalosard 

- Please merge with latest main the branch looks stale to me
- After merge with latest you must adapt the documentation regarding the playground component (see other docs e.g. buttons.md)
- Dropdown without icon looks broken
image

**Comment by goncalosard** [comment]
> @goncalosard
> 
> * Please merge with latest main the branch looks stale to me
> * After merge with latest you must adapt the documentation regarding the playground component (see other docs e.g. buttons.md)
> * Dropdown without icon looks broken
> 
> image

@danielleroux 

Added all the requests you made. Please check if its ok.

Build failed because component-doc.json, add your file.

**Review by danielleroux — APPROVED** [review]
lgtm

**Review by nuke-ellington — APPROVED** [review]
lgtm

**Pull request merged**